From 459f8ba53f841c3777f727e9ea0128ee08b3ed43 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Thu, 20 Apr 2017 18:54:59 +0100
Subject: [PATCH] Support a stateless configuration by default, using
 PulseAudio

This configuration will include the vendor-directory inside the defaults
tree as a configuration source, which in turn contains a default sane
configuration for alsa to hook via PulseAudio.

Users can still override this configuration through the typical /etc/ and
~/ paths.

Signed-off-by: Ikey Doherty <michael.i.doherty@intel.com>
---
 src/conf/Makefile.am |  5 ++++-
 src/conf/alsa.conf   |  1 +
 src/conf/asound.conf | 12 ++++++++++++
 3 files changed, 17 insertions(+), 1 deletion(-)
 create mode 100644 src/conf/asound.conf

diff --git a/src/conf/Makefile.am b/src/conf/Makefile.am
index 4f32662..0476a46 100644
--- a/src/conf/Makefile.am
+++ b/src/conf/Makefile.am
@@ -17,8 +17,11 @@ cfg_files += smixer.conf
 endif
 endif
 
-EXTRA_DIST = $(cfg_files)
+EXTRA_DIST = $(cfg_files) asound.conf
 
 alsaconfigdir = @ALSA_CONFIG_DIR@
 alsadir = $(alsaconfigdir)
 alsa_DATA = $(cfg_files)
+
+alsadefaultsdir = $(datadir)/defaults/alsa
+alsadefaults_DATA = asound.conf
diff --git a/src/conf/asound.conf b/src/conf/asound.conf
new file mode 100644
index 0000000..0da9682
--- /dev/null
+++ b/src/conf/asound.conf
@@ -0,0 +1,12 @@
+pcm.pulse {
+    type pulse
+}
+ctl.pulse {
+    type pulse
+}
+pcm.!default {
+    type pulse
+}
+ctl.!default {
+    type pulse
+}
\ No newline at end of file
-- 
2.12.2

--- alsa-lib-1.1.7/src/conf/alsa.conf~	2018-10-16 08:08:35.000000000 +0000
+++ alsa-lib-1.1.7/src/conf/alsa.conf	2018-11-01 01:27:00.299847974 +0000
@@ -8,6 +8,7 @@
 	{
 		func load
 		files [
+			"/usr/share/defaults/alsa/asound.conf"
 			"/etc/alsa/conf.d"
 			"/etc/asound.conf"
 			"~/.asoundrc"
